### PR TITLE
Convert shared Bridge class to JS

### PR DESCRIPTION
**Pull request by robertknight**
Convert the Bridge class & tests to JS, with the following additional changes:

- Removed use of Angular in the tests, since the class is not part of the sidebar code and doesn't have any injected dependencies anyway.
- Improved & extended API documentation.

For future reference, the way I'm approaching each of these conversions is:

1. Run the latest version of [decaffeinate](https://github.com/decaffeinate/decaffeinate) on the source.
2. Perform any manual cleanup required (fix lint warnings, convert `let` => `var` for consistency, remove unnecessary `return` statements).
3. Run the tests & commit.
4. Run decaffeinate on the tests.
5. Perform any manual cleanup of the tests.
6. Commit the converted tests.
7. Do one further pass over the code in case there is any dead code which can be removed, obsolete comments etc.

### CODE REVIEW

**Comment by robertknight** [comment]
CC @JCCR - Does this change conflict with any of the work you are doing?

**Comment by JCCR** [comment]
@robertknight It's fine, we just have some small pending changes planned for it but you can go ahead and merge this. I like it better if we get our changes in with JS from the start instead of it landing as converted coffeescript.

File: src/shared/bridge.js [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+   * @param {string} origin - The origin of the document in `source`.
+   * @param {string} token
+   * @return {RPC} - Channel for communicating with the window.
+   */
+  createChannel(source, origin, token) {
+    var channel = null;
+    var connected = false;
+
+    var ready = () => {
+      if (connected) { return; }
```
Comment (seanh) [inline src/shared/bridge.js]: :+1: I think this looks nice on one line

File: src/shared/bridge.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+   * Create a communication channel between this window and `source`.
+   *
+   * The created channel is added to the list of channels which `call`
+   * and `on` send and receive messages over.
+   *
+   * @param {Window} source - The source window.
+   * @param {string} origin - The origin of the document in `source`.
+   * @param {string} token
+   * @return {RPC} - Channel for communicating with the window.
+   */
```
Comment (seanh) [inline src/shared/bridge.js]: The new docstrings are great

**Pull request merged**